.devcontainer/cesm-build/devcontainer.json

```diff
@@ -0,0 +1,24 @@
+{
+    "containerEnv": {
+        "CIME_MODEL": "cesm",
+        "CIME_MACHINE": "docker",
+        "SKIP_CIME_UPDATE": "true",
+        "SKIP_MODEL_SETUP": "true"
+    },
+    "build": {
+        "dockerfile": "../../docker/Dockerfile",
+        "target": "base"
+    },    "workspaceFolder": "/src/cime",
+    "workspaceMount": "source=${localWorkspaceFolder}/..,target=/src,type=bind",
+    "postAttachCommand": "/entrypoint.sh bash",
+    "customizations": {
+        "vscode": {
+            "extensions": [
+                "charliermarsh.ruff"
+            ]
+        }
+    },
+    "features": {
+        "ghcr.io/devcontainers/features/git:1": {}
+    }
+}
```

Comment:
Why are these container changes part of this?

Reply:
I actually don't care if you keep it in this PR as long as we understand why it's there and you add it to the description.

Reply:
Probably best to pull it out of this PR. I'll be making a more testing focused PR soon, it'll fit in better there and I can provide a full explanation.